### PR TITLE
140041 - Bess bs save bespoke exchange set on fss

### PR DESCRIPTION
**Pull request by chaital14503**
This PR consists changes from [PBI140041](https://dev.azure.com/ukhydro/Exchange%20Set%20Service/_workitems/edit/140041)

### CODE REVIEW

File: UKHO.PeriodicOutputService/UKHO.BESS.BuilderService.UnitTests/Services/BuilderServiceTests.cs [hunk 1]
```diff
@@ -151,6 +175,34 @@ public async Task WhenTypeIsBase_ThenPostProductIdentifiersEndpointIsCalledAndBe
...
+            && call.GetArgument<LogLevel>(0) == LogLevel.Information
+            && call.GetArgument<EventId>(1) == EventIds.BESBatchCreationStarted.ToEventId()
+            && call.GetArgument<IEnumerable<KeyValuePair<string, object>>>(2)!.ToDictionary(c => c.Key, c => c.Value)["{OriginalFormat}"].ToString() == "BES batch creation started at {DateTime} | _X-Correlation-ID: {CorrelationId}"
+            ).MustHaveHappenedOnceExactly();
+
+            A.CallTo(fakeLogger).Where(call =>
+            call.Method.Name == "Log"
+            && call.GetArgument<LogLevel>(0) == LogLevel.Information
+            && call.GetArgument<EventId>(1) == EventIds.BESBatchCreationCompleted.ToEventId()
+            && call.GetArgument<IEnumerable<KeyValuePair<string, object>>>(2)!.ToDictionary(c => c.Key, c => c.Value)["{OriginalFormat}"].ToString() == "BES batch {besBatchId} is created and added to FSS with status: {isCommitted} at {DateTime} | _X-Correlation-ID: {CorrelationId}"
```
Comment (Nadeem14456) [inline UKHO.PeriodicOutputService/UKHO.BESS.BuilderService.UnitTests/Services/BuilderServiceTests.cs:204]: rename BES to BESS wherever used. 
```suggestion
            && call.GetArgument<IEnumerable<KeyValuePair<string, object>>>(2)!.ToDictionary(c => c.Key, c => c.Value)["{OriginalFormat}"].ToString() == "BESS batch {bessBatchId} is created and added to FSS with status: {isCommitted} at {DateTime} | _X-Correlation-ID: {CorrelationId}"
```

File: UKHO.PeriodicOutputService/UKHO.BESS.BuilderService.UnitTests/Services/BuilderServiceTests.cs [hunk 2]
```diff
@@ -219,8 +281,29 @@ public async Task WhenTypeIsUpdateOrChange_ThenGetProductVersionsEndpointIsCalle
...
```
Comment (Nadeem14456) [inline UKHO.PeriodicOutputService/UKHO.BESS.BuilderService.UnitTests/Services/BuilderServiceTests.cs:306]: rename BES to BESS wherever used.

File: UKHO.PeriodicOutputService/UKHO.BESS.BuilderService/Services/BuilderService.cs [hunk 3]
```diff
@@ -235,51 +224,73 @@ private void CreateZipFile(List<FssBatchFile> fileDetails, string downloadPath)
...
                     throw new Exception($"Creating zip file {file.FileName} failed at {DateTime.Now.ToUniversalTime()} | _X-Correlation-ID:{CommonHelper.CorrelationID}", ex);
                 }
             });
         }
 
-        private async Task<bool> CreateBessBatchAsync(string downloadPath, string fileExtension, Batch batchType)
+        private async Task<bool> CreateBessBatchAsync(string downloadPath, string fileExtension, ConfigQueueMessage configQueueMessage)
         {
             bool isCommitted;
+            string besBatchId;
```
Comment (Nadeem14456) [inline UKHO.PeriodicOutputService/UKHO.BESS.BuilderService/Services/BuilderService.cs:236]: rename BES to BESS wherever used.

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Enums/Batch.cs [hunk 4]
```diff
@@ -16,6 +16,6 @@ public enum Batch
         BesBaseZipBatch = 12,
         BesUpdateZipBatch = 13,
         BesChangeZipBatch = 14,
```
Comment (Nadeem14456) [inline UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Enums/Batch.cs:18]: rename BES to BESS wherever used.

File: UKHO.PeriodicOutputService/UKHO.BESS.BuilderService/Services/BuilderService.cs
Comment (Nadeem14456) [inline UKHO.PeriodicOutputService/UKHO.BESS.BuilderService/Services/BuilderService.cs:1]: reorganize read-only & const declaration.

**Review by Nadeem14456 — CHANGES_REQUESTED** [review]

File: UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Enums/Batch.cs [hunk 5]
```diff
@@ -74,7 +74,7 @@ public enum Batch
         EssEmptyBatch = 23,
 
         [Description("d0635e6c-81ae-4acb-9129-1a69f9ee58d2")]
-        EssEmptyBatchs57 = 24
+        BesEmptyBatch = 24
```
Comment (harshal11869) [inline UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Enums/Batch.cs:77]: Use Bess to make it consistent in mock as well wherever it get used.
```suggestion
        BeesEmptyBatch = 24
```

File: azure-pipelines.yml [hunk 6]
```diff
@@ -606,143 +606,143 @@ stages:
...
-                - task: AzureCLI@2
-                  displayName: "Swap Storage Container Configuration"
-                  condition: always()
-                  inputs:
-                    azureSubscription: "Exchange-Set-Service-Dev-A-008-02"
-                    scriptType: 'pscore'
-                    scriptLocation: 'scriptPath'
-                    scriptPath: "$(Build.SourcesDirectory)/devterraformartifact/swap_container_config.ps1"
-                    arguments: '-containername $(BessStorageConfiguration.ContainerName_PostFT) -resourcegroup $(RGName) -webappname $(WEB_APP_NAME)'
+      # - job: POSFunctionalTests
```
Comment (harshal11869) [inline azure-pipelines.yml:609]: FTs are commented ?

**Review by harshal11869 — CHANGES_REQUESTED** [review]

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Logging/EventIds.cs [hunk 7]
```diff
@@ -783,7 +783,22 @@ public enum EventIds
...
+
+        /// <summary>
+        /// 809157 - BES batch creation completed
+        /// </summary>
+        BESBatchCreationCompleted = 809157,
+
+        /// <summary>
+        /// 809158 - BES batch creation failed
+        /// </summary>
+        BESBatchCreationFailed = 809158
```
[the remark this answers is not in the document]
Comment (chaital14503) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Logging/EventIds.cs]: Resolved

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Services/FssService.cs [hunk 8]
```diff
@@ -181,6 +181,31 @@ public async Task<bool> DownloadFileAsync(string fileName, string fileLink, long
...
+                return createBatchResponse.BatchId;
+            }
+            else
+            {
+                _logger.LogError(EventIds.CreateBatchFailed.ToEventId(), "Request to create batch for {BatchType} in FSS failed | {DateTime} | StatusCode : {StatusCode} | _X-Correlation-ID : {CorrelationId}", batchType, DateTime.Now.ToUniversalTime(), httpResponse.StatusCode.ToString(), CommonHelper.CorrelationID);
+                throw new FulfilmentException(EventIds.CreateBatchFailed.ToEventId());
+            }
+        }
+
+
```
[the remark this answers is not in the document]
Comment (chaital14503) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Services/FssService.cs]: Resolved

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Services/FssService.cs [hunk 9]
```diff
@@ -552,22 +555,45 @@ private CreateBatchRequestModel CreateBatchRequestModel(Batch batchType)
...
+            CreateBatchRequestModel createBatchRequest;
+            List<KeyValuePair<string, string>> batchAttributes = new();
+
+            foreach (Tag tag in configQueueMessage.Tags)
+            {
+                batchAttributes.Add(new KeyValuePair<string, string>(tag.Key, tag.Value));
+            }
+
+            createBatchRequest = new()
+            {
```
[the remark this answers is not in the document]
Comment (chaital14503) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Services/FssService.cs]: Resolved

File: UKHO.PeriodicOutputService/UKHO.BESS.BuilderService.UnitTests/Services/BuilderServiceTests.cs (hunk 1 above)
Comment (chaital14503) [reply UKHO.PeriodicOutputService/UKHO.BESS.BuilderService.UnitTests/Services/BuilderServiceTests.cs]: Resolved

File: UKHO.PeriodicOutputService/UKHO.BESS.BuilderService.UnitTests/Services/BuilderServiceTests.cs (hunk 2 above)
Comment (chaital14503) [reply UKHO.PeriodicOutputService/UKHO.BESS.BuilderService.UnitTests/Services/BuilderServiceTests.cs]: Resolved

File: UKHO.PeriodicOutputService/UKHO.BESS.BuilderService/Services/BuilderService.cs (hunk 3 above)
Comment (chaital14503) [reply UKHO.PeriodicOutputService/UKHO.BESS.BuilderService/Services/BuilderService.cs]: Resolved

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Enums/Batch.cs (hunk 4 above)
Comment (chaital14503) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Enums/Batch.cs]: Resolved

File: UKHO.PeriodicOutputService/UKHO.BESS.BuilderService/Services/BuilderService.cs
Comment (chaital14503) [reply UKHO.PeriodicOutputService/UKHO.BESS.BuilderService/Services/BuilderService.cs:1]: Resolved

File: UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Enums/Batch.cs (hunk 5 above)
Comment (chaital14503) [reply UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Enums/Batch.cs]: Resolved

File: azure-pipelines.yml (hunk 6 above)
Comment (chaital14503) [reply azure-pipelines.yml]: Resolved

Comment (Nadeem14456) [reply azure-pipelines.yml]: > FTs are commented ?

this was temporary change. reverted.

**Review by Nadeem14456 — APPROVED** [review]
LGTM.

**Review by harshal11869 — APPROVED** [review]
Looks Good.

**Review by Arit14975 — APPROVED** [review]
LGTM from testing side.

**Review by DanOwenUKHO — APPROVED** [review]

**Pull request merged**